proto/tendermint/abci/types.proto

```diff
@@ -21,21 +21,26 @@ import "gogoproto/gogo.proto";
 
 message Request {
   oneof value {
-    RequestEcho               echo                 = 1;
-    RequestFlush              flush                = 2;
-    RequestInfo               info                 = 3;
-    RequestInitChain          init_chain           = 4;
-    RequestQuery              query                = 5;
-    RequestBeginBlock         begin_block          = 6;
-    RequestCheckTx            check_tx             = 7;
-    RequestDeliverTx          deliver_tx           = 8;
-    RequestEndBlock           end_block            = 9;
-    RequestCommit             commit               = 10;
-    RequestListSnapshots      list_snapshots       = 11;
-    RequestOfferSnapshot      offer_snapshot       = 12;
-    RequestLoadSnapshotChunk  load_snapshot_chunk  = 13;
-    RequestApplySnapshotChunk apply_snapshot_chunk = 14;
+    RequestEcho                echo                  = 1;
+    RequestFlush               flush                 = 2;
+    RequestInfo                info                  = 3;
+    RequestInitChain           init_chain            = 4;
+    RequestQuery               query                 = 5;
+    RequestBeginBlock          begin_block           = 6 [deprecated = true];
```

Comment:
I don't know what the plan is, but there are a couple of considerations:

1. During the transition, keeping the fields allows us to make the software see that they're set and report an appropriate diagnostic ("your software is out of date!" rather than "you omitted this required input!")
2. If any of these messages land in block data, we _might_ want to keep those around longer, so that we could handle soft migrations.
3. But if not (2), then we should probably plan to eventually drop them entirely, perhaps in the following release.

Reply:
I think Michael's option 3 is what we will end up doing, as those are not used in the block structure as such. But for the time being, until we change de logic that uses then to use the new APIs, let's leave them like that so that we don't break the build.
Regarding 17 and 18, if you take a look at the third commit of this PR in isolation (f9b42ecc) you'll see the point for those.